homeassistant/components/home_plus_control/api.py

```diff
@@ -0,0 +1,352 @@
+
+from .const import (
+    API,
+    CONF_MODULE_STATUS_UPDATE_INTERVAL,
+    CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL,
+    CONF_PLANT_UPDATE_INTERVAL,
+    CONF_SUBSCRIPTION_KEY,
+    DEFAULT_UPDATE_INTERVALS,
+    DOMAIN,
+    PLANT_URL,
+)
+
+
+async def update_api_refresh_intervals(hass, config_entry):
+    """Update the refresh data intervals based on the config entry values."""
+    api = hass.data[DOMAIN][config_entry.entry_id][API]
+    api.update_refresh_intervals()
+
+
+class HomePlusControlAsyncApi(HomePlusOAuth2Async):
+    """Legrand Home+ Control object that interacts with the OAuth2-based API of the provider.
+
+    This API is bound the HomeAssistant Config Entry that corresponds to this component.
+
+    Attributes:.
+        hass (HomeAssistant): HomeAssistant core object.
+        config_entry (ConfigEntry): ConfigEntry object that configures this API.
+        implementation (AbstractOAuth2Implementation): OAuth2 implementation that handles AA and
+                                                       token refresh.
+        logger (Logger): Logger of the object.
+        switches (dict): Dictionary of HomePlusControl switches indexed by their unique ID.
+        switches_to_remove (dict): Dictionary of the HomePlusControl switches that are to be
+                                   removed from HomeAssistant, indexed by their unique ID.
+    """
+
+    def __init__(
+        self,
+        hass: core.HomeAssistant,
+        config_entry: config_entries.ConfigEntry,
+        implementation: config_entry_oauth2_flow.AbstractOAuth2Implementation,
+    ):
+        """Initialize the HomePlusControlAsyncApi object.
+
+        Initialize the authenticated API for the Legrand Home+ Control component.
+
+        Args:.
+            hass (HomeAssistant): HomeAssistant core object.
+            config_entry (ConfigEntry): ConfigEntry object that configures this API.
+            implementation (AbstractOAuth2Implementation): OAuth2 implementation that handles AA
+                                                           and token refresh.
+        """
+        self.hass = hass
+        self.config_entry = config_entry
+        self.implementation = implementation
+        self._domain = DOMAIN
+        self._plants = {}
+        self._switches = {}
+        self._switches_to_remove = {}
+
+        self._last_check = {
+            CONF_PLANT_UPDATE_INTERVAL: time.monotonic(),
+            CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL: -1,
+            CONF_MODULE_STATUS_UPDATE_INTERVAL: -1,
+        }
+
+        self._refresh_intervals = {}
+        self.update_refresh_intervals()
+
+        # Create the API authenticated client - external library
+        super().__init__(
+            client_id=self.config_entry.data[CONF_CLIENT_ID],
+            client_secret=self.config_entry.data[CONF_CLIENT_SECRET],
+            subscription_key=self.config_entry.data[CONF_SUBSCRIPTION_KEY],
```

Comment:
The subscription key is available on the implementation instance. If we put it there we should not store it in config entry data.